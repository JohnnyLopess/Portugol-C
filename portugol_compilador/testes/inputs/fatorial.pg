funcao inteiro fat(inteiro n)
    se n == 0 entao
        retorne 1;
    senao
        retorne n * fat(n - 1);
    fimse
fimfuncao

inicio
    escreva(fat(5));
fim