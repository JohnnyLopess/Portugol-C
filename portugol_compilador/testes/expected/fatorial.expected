#include <stdio.h>
#include <string.h>

int fat(int n) {
    if (n == 0) {
        return 1;
    } else {
        return n * fat(n - 1);
    }
}

int main() {
    printf("%d\n", fat(5));
    return 0;
}
